### PR TITLE
test/e2e: Collect CatalogSource testing artifacts

### PR DESCRIPTION
**Pull request by timflannagan**
Update the test/e2e/collect-ci-artifacts.sh bash script and extend the
list of resources gathered to also create a snapshot of CatalogSource
resources in the test namespace.

**Description of the change:**

**Motivation for the change:**

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/doc`
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

File: test/e2e/collect-ci-artifacts.sh [hunk 1]
```diff
@@ -24,3 +25,8 @@ for command in "${commands[@]}"; do
     COMMAND_OUTPUT_FILE=${TEST_ARTIFACTS_DIR}/${command// /_}
     kubectl -n ${TEST_NAMESPACE} ${command} >> "${COMMAND_OUTPUT_FILE}"
 done
+
+echo "Deleting any empty files in the ${TEST_ARTIFACTS_DIR} directory"
+# Note: delete any files less than 83 kilobytes as this should account for
+# any empty files, and files that contain zero length arrays of `kind: List`.
+find "${TEST_ARTIFACTS_DIR}" -type f -size -83k -delete
```
Comment (timflannagan) [inline test/e2e/collect-ci-artifacts.sh:32]: Hmm I realize we should probably torch empty directories as well. It looks like find also has support for identifying empty directories so it should just be another command.

File: test/e2e/collect-ci-artifacts.sh [hunk 2]
```diff
@@ -11,6 +11,7 @@ set -o errexit
 mkdir -p "${TEST_ARTIFACTS_DIR}"
 
 commands=()
+commands+=("get catalogsources -o yaml")
```
Comment (njhale) [inline test/e2e/collect-ci-artifacts.sh:14]: nit: if the string doesn't need to be interpolated, use single quotes.

**Review by njhale — APPROVED** [review]
/approve

**Pull request merged**